server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -1478,6 +1483,12 @@ public boolean equals(Object o) {
         if (isSystem != that.isSystem) {
             return false;
         }
+        if (reshardingMetadata == null && that.reshardingMetadata != null) {
```

Comment:
Objects.equals(reshardingMetadata, that.reshardingMetadata)

Reply:
Done in 3f1948471f2e00915fa670354b8f357c0a87da58